### PR TITLE
Add WebXR Hit Test reference docs - pt 4

### PR DESCRIPTION
**Pull request by Elchi3**
Part 4 documents `XRHitTestResult` and its two methods `getPose()` and `createAnchor()`.

Specs:
- https://immersive-web.github.io/hit-test/#xr-hit-test-result-interface
- https://immersive-web.github.io/anchors/#anchor-creation

### CODE REVIEW

File: files/en-us/web/api/xrhittestresult/index.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- {{domxref("XRHitTestResult.createAnchor()")}}
+  - : Creates an {{domxref("XRAnchor")}} from the hit test result.
+- {{domxref("XRHitTestResult.getPose()")}}
+  - : Returns the {{domxref("XRPose")}} of the hit test result relative to the given base space.
+
+## Examples
+
+### Getting `XRHitTestResult` objects within the frame loop
+
+To get hit test results, you need to provide a {{domxref("XRHitTestSource")}}. See {{domxref("XRSession.requestHitTestSource()")}} for more information.
```
Comment (jpmedley) [inline files/en-us/web/api/xrhittestresult/index.md:33]:
```suggestion
In addition to showing `XRHitTestResult` within a frame loop, this example demonstrates a few things you must do before requesting this object. While setting up the session, specify `"hit-test"` as one of the `requiredFeatures`. Next, call {{domxref("XRSession.requestHitTestSource()")}} with the desired references. (Obtain this by calling {{domxref("XRSession.requestReferenceSpace()")}}.) This returns a {{domxref("XRHitTestSource")}}. That you will use in the frame loop to get `XRHitTestResult` objects. 
```

File: files/en-us/web/api/xrhittestresult/index.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+let hitTestResults = xrFrame.getHitTestResults(hitTestSource);
+
+if (hitTestResults.length > 0) {
+  let pose = hitTestResults[0].getPose(referenceSpace);
+}
+```
+
+### Creating an anchor from a hit test result
+
+Once you found intersections on real-world surfaces using hit testing, you can create an {{domxref("XRAnchor")}} to attach a virtual object to that location.
```
Comment (jpmedley) [inline files/en-us/web/api/xrhittestresult/index.md:71]:
```suggestion
Once you find intersections on real-world surfaces using hit testing, you can create an {{domxref("XRAnchor")}} to attach a virtual object to that location.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/xrhittestresult/createanchor/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Examples
+
+### Creating an anchor from a hit test result
+
+Once you found intersections on real-world surfaces by using hit testing, you can create an {{domxref("XRAnchor")}} to attach a virtual object to that location.
+
+```js
+hitTestResult.createAnchor().then((anchor) => {
+  // add anchored objects to the scene
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/web/api/xrhittestresult/createanchor/index.md:40]: I think it is more involved. I can try to follow-up and query an SME to help me on this (if you think this is mission-critical).

File: files/en-us/web/api/xrhittestresult/getpose/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- `baseSpace`
+  - : An {{domxref("XRSpace")}} to use as the base or origin for the purposes of computing the relative position and orientation.
+
+### Return value
+
+Returns an {{domxref("XRPose")}} object.
+
+## Examples
+
+### Getting the hit test result's pose
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/web/api/xrhittestresult/getpose/index.md:35]: Sentence case per https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide#page_titles

File: files/en-us/web/api/xrhittestresult/createanchor/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/xrhittestresult/createanchor/index.md:40]: I'll approve the PR, but I think we need to give some kind of handle the reader even if it's just a link to something like an article or a more robust example.

[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/xrhittestresult/createanchor/index.md:40]: I don't just check actual links. I make sure that conceptual links aren't broken either.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**